disperser/controller/dispatcher.go

```diff
@@ -489,12 +491,41 @@ func (d *Dispatcher) sendChunks(ctx context.Context, client clients.NodeClient,
 	return sig, nil
 }
 
-func (d *Dispatcher) updateBatchStatus(ctx context.Context, keys []corev2.BlobKey, status v2.BlobStatus) error {
-	for _, key := range keys {
-		err := d.blobMetadataStore.UpdateBlobStatus(ctx, key, status)
+func (d *Dispatcher) updateBatchStatus(ctx context.Context, batch *batchData, attestation *corev2.Attestation) error {
+	var multierr error
+	for i, cert := range batch.Batch.BlobCertificates {
+		blobKey := batch.BlobKeys[i]
+		if cert == nil || cert.BlobHeader == nil {
+			d.logger.Error("invalid blob certificate in batch")
+			err := d.blobMetadataStore.UpdateBlobStatus(ctx, blobKey, v2.Failed)
+			if err != nil {
+				multierr = multierror.Append(multierr, fmt.Errorf("failed to update blob status for blob %s to failed: %w", blobKey.Hex(), err))
+			}
+			continue
+		}
+
+		failed := false
+		for _, q := range cert.BlobHeader.QuorumNumbers {
+			if res, ok := attestation.QuorumResults[q]; !ok || res == 0 {
+				d.logger.Error("quorum result not found", "quorumID", q, "blobKey", blobKey.Hex())
+				failed = true
+				break
+			}
+		}
+
+		if failed {
+			err := d.blobMetadataStore.UpdateBlobStatus(ctx, blobKey, v2.INSUFFICIENT_SIGNATURES)
```

Comment:
So right now the insufficient signature state is for when a quorum has no attestations?

Reply:
Yeah
We can improve this by either considering the coding rate or the default confirmation threshold onchain, but this is a quick fix before further steps